### ISSUE TITLE
Enable Bluetooth support

### ISSUE THREAD

**Issue by robmckinnon** [emfcamp/badge-2024-software#13]
Can we please consider enabling Bluetooth support in the MicroPython firmware for the badge by default?

What are the arguments for or against this? 

Is it a power consumption issue, is there a way around that?

I'd like to write an app for the badge to use [BLE MIDI](https://midi.org/category/midi-connections/bluetooth-low-energy-ble-wireless-midi) to communicate wirelessly with BLE MIDI enabled synth and MIDI controller hardware. I imagine this would be easier to develop and share with others if Bluetooth was enabled by default.

In the [mpconfigboard.h](https://github.com/search?q=repo%3Aemfcamp%2Fbadge-2024-software%20mpconfigboard.h&type=code) file adding `#define MICROPY_PY_BLUETOOTH                (1)` enables Bluetooth support in the MicroPython firmware.

**Comment by MatthewWilkes** [comment]
@robmckinnon BLE support is already enabled, but the full Bluetooth stack isn't. Is that sufficient for you?

At https://github.com/emfcamp/badge-2024-software/blob/main/tildagon/mpconfigboard.cmake#L39 we include the following config file: https://github.com/micropython/micropython/blob/cfd5a8ea3a80930a79dbae6a6aff54587393d0b8/ports/esp32/boards/sdkconfig.ble

I can confirm that `bluetooth.BLE` is available on our test builds, and we've been able to view advertisements.  Does that cover your needs?

**Comment by fridgehead** [comment]
yeah it works for us too, using aioble to talk to a strip of LED's in our dome:
```

async def connectDevice(self):
        device = aioble.Device(aioble.ADDR_PUBLIC, "23:00:00:00:00:00")
        conn = await device.connect()
        try:
            connection = await device.connect(timeout_ms=2000)
        except asyncio.TimeoutError:
            print('Timeout')
        print("connected!")
        temp_service = await connection.service(bluetooth.UUID(0xafd0))
        temp_char = await temp_service.characteristic(bluetooth.UUID(0xafd1))
        print("offen")
        offbytes = b"\x5B\x0f\x00\xB5"
        await temp_char.write(offbytes, timeout_ms=1000)

```

**Comment by augeas** [comment]
BLE still seems somewhat dead:

This keels over with `OSError: [Errno 5] Input/output error` in the REPL...

```python

import aioble
import app``
import asyncio

class TildaMesh(app.App):
    def __init__(self):
        super().__init__()
        self.devices = []

    async def get_devices(self):
        devices = []
        async with aioble.scan(5000) as scanner:
            async for result in scanner:
                name = result.name()
                print(name)
                devices.append(name)

        return devices

    async def run(self, render_update):
        await render_update()

        self.devices = await self.get_devices()

        while True:
            await asyncio.sleep(1)
            print('ping...')
            await render_update()

    def draw(self, ctx):
        ctx.save()
        ctx.rgb(0, 0, 0).rectangle(-120, -120, 240, 240).fill()
        ctx.rgb(1, 0, 0).move_to(-80, 0).text("TildaMesh")
        ctx.restore()

__app_export__ = TildaMesh
```

**Comment by JonTheNiceGuy** [comment]
I've now lost the docs and code that Claude wrote for me at the time, but there's a hardware line shared between the LCD driver and the Bluetooth chip which means that you can't run both at the same time. Even if you revert the patch of the firmware to move the framebuffer back into the SPI RAM (where it was in the initial firmware), then you still can only ever get up to 3FPS on the LCD.

But it still requires the framebuffer to be in SPI RAM to prevent the crash.

**Comment by Jonty** [comment]
We've tracked this down, it'll be fixed by the upcoming ESP-IDF upgrade! https://github.com/emfcamp/badge-2024-software/pull/268#issuecomment-4262692910

**Comment by ChrisDick** [comment]
The fix is now in main and [v2.0.0-alpha.2](https://github.com/emfcamp/badge-2024-software/releases/tag/v2.0.0-alpha.2) , so closing this.
